documentation/src/developer_documentation/state_migration_spike.md

```diff
@@ -0,0 +1,139 @@
+## State migration spike 🛂
+
+### What is state migration?
+
+State migration is a process where the `StateTree` contents are transformed from
+an older form to a newer form.
+
+### Why do we need to migrate?
+
+Migration is required when the structure of the state changes. This happens when
+new fields are added or existing ones are modified. Migration is **not**
+required in case of new behaviour.
+
+In case of NV18, the `StateTree` changed from version 4 to version 5. See
```

Comment:
For the example of NV18, there's more than just the StateTree version changing. For example, the init actor is being [migrated](https://github.com/filecoin-project/go-state-types/blob/master/builtin/v10/migration/init.go#L32) as well